src/zenml/integrations/kubernetes/step_operators/kubernetes_step_operator.py

```diff
@@ -0,0 +1,226 @@
+
+        Args:
+            deployment: The pipeline deployment for which to get the builds.
+
+        Returns:
+            The required Docker builds.
+        """
+        builds = []
+        for step_name, step in deployment.step_configurations.items():
+            if step.config.step_operator == self.name:
+                build = BuildConfiguration(
+                    key=KUBERNETES_STEP_OPERATOR_DOCKER_IMAGE_KEY,
+                    settings=step.config.docker_settings,
+                    step_name=step_name,
+                )
+                builds.append(build)
+
+        return builds
+
+    def get_kube_client(
+        self, incluster: Optional[bool] = None
+    ) -> k8s_client.ApiClient:
+        if incluster is None:
+            incluster = self.config.incluster
+
+        if incluster:
+            kube_utils.load_kube_config(
+                incluster=incluster,
+            )
+            self._k8s_client = k8s_client.ApiClient()
+            return self._k8s_client
+
+        # Refresh the client also if the connector has expired
+        if self._k8s_client and not self.connector_has_expired():
+            return self._k8s_client
+
+        connector = self.get_connector()
+        if connector:
+            client = connector.connect()
+            if not isinstance(client, k8s_client.ApiClient):
+                raise RuntimeError(
+                    f"Expected a k8s_client.ApiClient while trying to use the "
+                    f"linked connector, but got {type(client)}."
+                )
+            self._k8s_client = client
+        else:
+            kube_utils.load_kube_config(
+                context=self.config.kubernetes_context,
+            )
+            self._k8s_client = k8s_client.ApiClient()
+
+        return self._k8s_client
+
+    @property
+    def _k8s_core_api(self) -> k8s_client.CoreV1Api:
+        """Getter for the Kubernetes Core API client.
+
+        Returns:
+            The Kubernetes Core API client.
+        """
+        return k8s_client.CoreV1Api(self.get_kube_client())
+
+    def launch(
+        self,
+        info: "StepRunInfo",
+        entrypoint_command: List[str],
+        environment: Dict[str, str],
+    ) -> None:
+        """Launches a step on Kubernetes.
+
+        Args:
+            info: Information about the step run.
+            entrypoint_command: Command that executes the step.
+            environment: Environment variables to set in the step operator
+                environment.
+        """
+
+        settings = cast(
+            KubernetesStepOperatorSettings, self.get_settings(info)
+        )
+        image_name = info.get_image(
+            key=KUBERNETES_STEP_OPERATOR_DOCKER_IMAGE_KEY
+        )
+
+        pod_name = f"{info.run_name}_{info.pipeline_step_name}"
+        pod_name = kube_utils.sanitize_pod_name(pod_name)
+
+        command = entrypoint_command[:3]
+        args = entrypoint_command[3:]
```

Comment:
I like this. We should probably also do this clear separation in the vertex step operator, which puts the whole `entrypoint_command` into the pod spec's command attribute. Just a suggestion though; both approaches are correct.